src/js/components/search/filters/timePeriod/AllFiscalYears.jsx

```diff
@@ -58,15 +40,12 @@ export default class AllFiscalYears extends React.Component {
             allSelected = false;
         }
 
-        this.setState({
-            selectedFY: arrayFY,
-            allFY: allSelected
-        });
+        this.props.saveSelected(arrayFY, allSelected);
     }
 
     saveAllYears() {
```

Comment:
There is a bug in this function. If some FY years are individual selected before the user clicks "All Years," the All Years checkbox is checked but all the individual year entries become unchecked. Subsequently, the All Years checkbox is always inverse from what is checked (checked if the other years are unchecked, unchecked if the other years are checked).

I don't think this function needs to consider what values are already checked. We know if the user is going from unchecked -> checked or vice versa via the `allFY` props. Given the user's action, we should just completely overwrite the `selectedFY` values with either the full list of FYs or an empty set. We already have both on-hand, so no merges should be necessary (esp since the merge operation still outputs a new instance per `immutability-js`).